### PR TITLE
Conan 2 Support

### PR DESCRIPTION
**Pull request by crsib**
This PR introduces the support for Conan 2, inherently breaking the support for Conan 1. The upgrade should happen automatically for "managed" Conan installation users.

Most of the work related to Conan 2 support happened within https://github.com/audacity/conan-recipes; the corresponding PR will be created after Qt is fully ported to Conan 2.

One of the key ideas behind the rework is to stop using `conan-center` remote. This remote is somewhat fragile because the required Conan version is handled on the client side only, which randomly breaks local builds if Conan is not fresh enough. For this reason, all required recipes are now served using the channel `audacity/stable`

The following libraries were updated:

* `zlib` to 1.2.13
* `libpng` to 1.6.39
* `expat` to 2.5.0
* `mpg123` to 1.31.2
* `wavpack` to 5.6.0
* `ogg` to 1.3.5
* `flac` to 1.4.2
* `libsndfile` to 1.0.31
* `vst3sdk` to 3.7.7

The update also has affected `breakpad` and `crashpad`.

From the Audacity side, the user can now specify the host and build profiles manually, effectively addressing #4672.

@dozzzzer this PR need some smoke testing, focusing on Import/Export, VST3, and crash reports.

---

Resolves: #4360 
Resolves: #4672
Resolves: #4156
Resolves: #4270

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Review by vsverchinsky — APPROVED** [review]

File: conan/conanfile.py [hunk 1]
```diff
@@ -179,7 +179,7 @@ class AudacityConan(ConanFile):
 
         AudacityDependency("rapidjson", "1.1.0"),
 
-        AudacityDependency("breakpad", "0.1"),
+        AudacityDependency("breakpad", "2023.01.27"),
```
[the remark this answers is not in the document]
Comment (crsib) [reply conan/conanfile.py:182]: I've tried to stick to a "real" tag for breakpad, so it is clear what we get :-)

**Comment by LWinterberg** [comment]
This was surprisingly painless, after a `pip install --upgrade conan` and deleting the old build folder, it just worked, including has_audiocom_upload and has_asio_support builds. 

However, in this build it seems like realtime effects are just always bypassed, whether they be internal, LV2, VST or VST3. It seems like they don't receive any data at all. Also, WASAPI recording seems broken to me, it alternates between "invalid sample rate" and "insufficient memory". I'm not sure if these issues got introduced here though, all I know is that it's a regression against stable.

**Comment by crsib** [comment]
Let me take a look.

**Comment by crsib** [comment]
I can observe both in the latest master. I have no VST3 effects available in realtime too. 

Small note about WASAPI:
1. Only recording fails for me.
2. The first time I press record - Audacity succeeds, but subsequent calls fail.

@dozzzzer appears that master is badly broken; please help with bisection.

**Comment by dozzzzer** [comment]
- As for the non-working effects: this was introduced by https://github.com/audacity/audacity/commit/35a0869c240631d235f05c0063f340ba3d1feee8, the corresponding issue is https://github.com/audacity/audacity/issues/4859
- As for the WASAPI recording issue, I can't reproduce it on either Win 11 VM or Win 10 laptop.

**Comment by dozzzzer** [comment]
Tested:

- Import/Export and playback of Audio (all supported formats + selected audio and multiple tracks), MIDI, Labels
- VST3: playback with an RT effect, destructive application, loading presets, import/export of presets, non-GUI mode, mixing-and-rendering
- Crash report dialog pops up when Audacity crashes

Not tested:

- plugin settings persistence (due to the https://github.com/audacity/audacity/issues/4848 which won't allow to re-open a project after it was saved)
- crash reports sending

**Pull request merged**